test/performance/shmem_perf_suite/bw_common.h

```diff
@@ -541,10 +547,34 @@ void static inline calc_and_print_results(double total_t, int len,
     /* base case: will be overwritten by collective if num_pes > 2 */
     pe_bw_sum = bw;
 
-    if(nPEs >= 2)
-        shmem_double_sum_to_all(&pe_bw_sum, &bw, nred_elements, start_pe,
+    if (metric_info.individual_report == 1) {
+        printf("Individual bandwith for PE %d is %10.2f\n", metric_info.my_node, pe_bw_sum);
+    }
+    
+    pe_time = total_t;
+    shmem_barrier(start_pe, stride, nPEs, red_psync);
+    if (nPEs >= 2) {
+        shmem_double_max_to_all(&pe_time_max, &pe_time, nred_elements, start_pe,
                                 stride, nPEs, pwrk,
                                 red_psync);
+    }
+
+    /* calculating bandwidth based on the highest time taken across all PEs */
+    if (pe_time_max > 0 ) {
+#ifdef ENABLE_OPENMP
+        bw = (len * metric_info.midpt / 1.0e6 * metric_info.window_size * metric_info.trials *
+                (double)metric_info.nthreads) / (pe_time_max / 1.0e6);
+#else
+        bw = (len * metric_info.midpt / 1.0e6 * metric_info.window_size * metric_info.trials) /
+                (pe_time_max / 1.0e6);
+#endif
+    }
+
+    /* 2x as many messages/bytes at once for bi-directional */
+    if(metric_info.type == BI_DIR)
+        bw *= 2.0;
```

Comment:
Would it be better to multiply `len` by 2.0 above since it's twice the data volume for bi-directional?  This should yield the same result, but may be easier to understand.